extensions/blocks/map/edit.js

```diff
@@ -185,6 +212,51 @@ class MapEdit extends Component {
 						] }
 					/>
 					<PanelBody title={ __( 'Map Settings', 'jetpack' ) }>
+						<BaseControl
+							label={ __( 'Height in pixels' ) }
+							id={ `block-jetpack-map-height-input-${ instanceId }` }
+						>
+							<input
+								type="number"
+								id={ `block-jetpack-map-height-input-${ instanceId }` }
+								onChange={ event => {
+									let height = parseInt( event.target.value, 10 );
+									if ( isNaN( height ) ) {
+										// Set map height to default size and input box to empty string
+										height = null;
+									} else if ( height < 400 ) {
+										// Set map height to minimum size
+										height = 400;
+									}
+									setAttributes( {
+										mapHeight: height,
+									} );
+									setTimeout( this.mapRef.current.sizeMap, 0 );
```

Comment:
You could do this in `componentDidUpdate`, [example](https://github.com/Automattic/jetpack/blob/bf52e73459946042c4e587a2acbe9f1d643b6a4e/extensions/blocks/tiled-gallery/layout/mosaic/index.js#L29) more cleanly than using timeout.

Reply:
I copied this from the existing usage, which comes from when the map block was originally built (fc7596995ad368bbc6f88aa35afe0ef3ede18433). Reading through the original PRs (Automattic/wp-calypso#27836, Automattic/wp-calypso#28498), I couldn't see any discussion of why it was done this way.

@jeffersonrabb: I know it was a long time ago, but I don't suppose you recall if there was a specific reason for the `setTimeout()` call here, rather than using `componentDidUpdate()`?

Reply:
> @jeffersonrabb: I know it was a long time ago, but I don't suppose you recall if there was a specific reason for the `setTimeout()` call here, rather than using `componentDidUpdate()`?

No idea 😬. I added the line in the initial commit so there's no useful commit message. Given that there's no `componentDidUpdate` in `MapEdit` it's probably safe to assume the approach just didn't occur to me!

Reply:
I'm going to leave it as is for now, since it'd be a bit weird to only fix some, but not this others, and this PR has been going on for long enough as is. 🙂